lib/shared/pages/change_max_amount_bottom_sheet.dart

```diff
@@ -0,0 +1,218 @@
+    }
+    setState(() {
+      amount.text = (int.parse(amount.text) + 5).toString();
+    });
+  }
+
+  void decreaseAmount() {
+    if (int.parse(amount.text) == 0) {
+      return;
+    }
+    setState(() {
+      amount.text = (int.parse(amount.text) - 5).toString();
+    });
+  }
+
+  @override
+  void initState() {
+    amount = TextEditingController(text: widget.maxAmount.toString());
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    final locals = context.l10n;
+    final size = MediaQuery.sizeOf(context);
+    return BlocConsumer<PersonalInfoEditBloc, PersonalInfoEditState>(
+      listener: (context, state) {
+        if (state.status == PersonalInfoEditStatus.success) {
+          context
+              .read<AuthCubit>()
+              .refreshUser()
+              .whenComplete(() => context.pop());
+        }
+      },
+      builder: (context, state) {
+        return BottomSheetLayout(
+          title: Text(locals.giveLimit),
+          child: Column(
+            children: [
+              const SizedBox(
+                height: 20,
+              ),
+              Text(
+                locals.amountLimit,
+                style: Theme.of(context).textTheme.bodyLarge,
+                textAlign: TextAlign.center,
+              ),
+              SizedBox(
+                height: size.height * 0.3,
+              ),
+              Row(
+                mainAxisAlignment: MainAxisAlignment.spaceAround,
+                children: [
+                  IconButton.filled(
+                    onPressed: decreaseAmount,
+                    icon: const Icon(
+                      FontAwesomeIcons.circleMinus,
+                    ),
+                    color: AppTheme.softenedGivtPurple,
+                    iconSize: size.width * 0.1,
+                  ),
+                  Container(
+                    decoration: const BoxDecoration(
+                      border: Border(
+                        bottom: BorderSide(
+                          color: AppTheme.givtLightGreen,
+                          width: 7,
+                        ),
+                      ),
+                    ),
+                    child: Row(
+                      children: [
+                        Icon(
+                          widget.icon,
+                          size: size.width * 0.04,
+                        ),
+                        const SizedBox(
+                          width: 20,
+                        ),
+                        SizedBox(
+                          width: size.width * 0.15,
+                          child: TextFormField(
+                            controller: amount,
+                            keyboardType: TextInputType.number,
+                            onChanged: (value) {
+                              if (value.isEmpty) {
+                                amount.text = '0';
+                              }
+                              setState(() {});
```

Comment:
:)

Reply:
Yeah, didn't know how make it prettier to enable the button